packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -163,10 +163,23 @@ export default class ClassTransformer {
     //
     body.push(t.returnStatement(t.cloneNode(this.classRef)));
 
-    const container = t.arrowFunctionExpression(
-      closureParams,
-      t.blockStatement(body, [t.directive(t.directiveLiteral("use strict"))]),
-    );
+    const parent = this.path.findParent((path) => path.isProgram() || path.isBlockStatement());
```

Comment:
wont it check only the nearest parent? we can have really deep tree and I guess we should check for something like:
```js
const strictParent = this.path.findParent((path) => {
  if (!path.isProgram() && !path.isBlockStatement()) {
    return false
  }

  return path.node.directives.some(directive => directive.value.value === 'use strict')
});
```

Reply:
Yes that absolutely right! Thanks @Andarist